### PR TITLE
Fake selection

### PR DESCRIPTION
**Pull request by raffij**
**V1 - Can CTRL+A to Copy all content**
- [x] Implement Selection Handler.
- [x] Block level selection.
- [x] Feature flag additional functionality.
- [x] Cross browser fixes.
- [x] Add tests.

### CODE REVIEW

File: src/blocks/image.js [hunk 1]
```diff
@@ -43,5 +43,12 @@ module.exports = Block.extend({
         }
       );
     }
+  },
+
+  asClipboardHTML: function() {
```
Comment (callum) [inline src/blocks/image.js]: 👍

File: src/blocks/tweet.js [hunk 2]
```diff
@@ -100,5 +100,12 @@ module.exports = Block.extend({
   onDrop: function(transferData){
     var url = transferData.getData('text/plain');
     this.handleTwitterDropPaste(url);
+  },
+
+  asClipboardHTML: function() {
+    var data = this.getBlockData();
+    if (!data.status_url) return;
+
+    return `<p>${data.status_url}</p>`;
```
Comment (callum) [inline src/blocks/tweet.js]: Isn't there more data that could be used here to include the tweet text as well?

File: src/blocks/video.js [hunk 3]
```diff
@@ -72,6 +74,13 @@ module.exports = Block.extend({
   onDrop: function(transferData){
     var url = transferData.getData('text/plain');
     this.handleDropPaste(url);
+  },
+
+  asClipboardHTML: function() {
+    var data = this.getBlockData();
+    var source = this.providers[data.source];
+    var src = source.url(data.remote_id);
+    return `<p>${src}</p>`;
```
Comment (callum) [inline src/blocks/video.js]: Same here - couldn't the HTML be the iframe?

File: src/blocks/tweet.js (hunk 2 above)
Comment (ilyaporopudas) [reply src/blocks/tweet.js]: I don't think we have access to the tweet text.

File: src/blocks/video.js (hunk 3 above)
Comment (ilyaporopudas) [reply src/blocks/video.js]: As in, fetching the source code of the URL loaded in the `iframe`? I actually like this approach, as it sort of matches how I write articles even outside of Sir Trevor.

File: src/blocks/tweet.js (hunk 2 above)
Comment (raffij) [reply src/blocks/tweet.js]: I'm going to do some further testing with word and google docs to see what the best way of include complex blocks. I was talking with Callum and he gave some ideas I'm going to look at.

File: src/blocks/video.js (hunk 3 above)
Comment (raffij) [reply src/blocks/video.js]: @ilyaporopudas this comes from something i was talking to @callum about regarding the different reasons for copy / paste. At the moment i've built it for export to another system, but we'll want to include a text representation and also an internal version for when pasting back into sir-trevor.

Comment (ilyaporopudas) [reply src/blocks/video.js]: 👍

**Pull request merged**